src/main/scala/nl/knaw/dans/api/sword2/DepositHandler.scala

```diff
@@ -160,6 +166,38 @@ object DepositHandler {
     }
   }
 
+  private def resolveFetchItems(bagitDir: File)(implicit id: String): Try[Unit] = Try {
+    log.debug(s"[$id] Checking fetch.txt")
+    getFetchTxt(bagitDir).foreach(fetchText =>
+      { log.debug(s"[$id] Resolving files in fetch.txt")
+        fetchText.asScala.foreach(item =>
+        getUrl(item.getUrl).foreach(url =>
+          Using.urlInputStream(url).foreach(src =>
+            Files.copy(src, Paths.get(bagitDir.getAbsolutePath, item.getFilename)))))
+      }
+    )
+  }
+
```

Comment:
_getFetchTxt_ returns type Option[FetchTxt]. In this case _foreach_ gives the _FetchTxt_ if it is present and valid.  Scala prefers _foreach_ here over _map_.  I have understood that this is a common way in Scala (to handle either with _map_ or with  _foreach_ an Option type.

If the url is syntactically incorrect or when it can't be dereferenced, the client gets an error message "Invalid url in Fetch Items ($url)".  In the log the exact cause of the error is shown.